### PR TITLE
Fixes #33 fix exports non package

### LINKED ISSUE #33: get rid of "catkin_package() exports non-package include path"
`moveit_core` currently exports the header file "devel/include/moveit/version.h".
Because this is not a path specifically for `moveit_core`, `catkin_lint .` in the `moveit_core` source folder (rightfully) complains:

> moveit_core: CMakeLists.txt(77): error: catkin_package() exports non-package include path

The file should probably be generated in "devel/include/moveit_core/moveit/version.h" (using appropriate catkin-variables for the prefix).

### PR DESCRIPTION
**Pull request by AndriyPt**
This is fix for https://github.com/ros-planning/moveit/issues/33. Unfortunately catkin_lint requires only files from package directory or sub-directories to be in the catkin_package() include paths according to this code https://github.com/fkie/catkin_lint/blob/master/src/catkin_lint/linter.py#L112-L114. So adding creation of the sub-directory in the version folder for version.h file and line in .gitignore to avoid accidental commit.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -49,4 +49,7 @@ qtcreator-*
 CATKIN_IGNORE
 
 # Continous Integration
-.moveit_ci
\ No newline at end of file
+.moveit_ci
+
+# Autogenerated version file
+/moveit_core/version/moveit
```
Comment (davetcoleman) [inline .gitignore]: catkin is generating build artifacts in your source tree? this seems unlikely

**Comment by davetcoleman** [comment]
If i understand this correctly - you are generating a file in the source space which you should not do. I'll let @v4hn look at this

**Comment by davetcoleman** [comment]
@wjwwood is going to look at this

**Comment by wjwwood** [comment]
@AndriyPt Thanks for the contribution! I believe the right thing to do here would be to generate the file into the devel space (as it is now), or into the build space, but into a folder that is already on the include path and that is already being installed. I'm going to try out a patch I have in mind and post it here to see what everyone thinks.

I have to agree with @davetcoleman that generating the file into the source space goes against our best practices.

**Pull request closed without merging**